### ISSUE TITLE
RuntimeError: _convert_weight_to_int4pack_cpu : expect weight to be kByte

### ISSUE THREAD

**Issue by dvrogozh** [huggingface/optimum-quanto#274]
I am trying quanto library from head of the main branch (currently at https://github.com/huggingface/optimum-quanto/commit/601dc193ce0ed381c479fde54a81ba546bdf64d1) against self-built pytorch main branch (upcoming 2.5, I am at https://github.com/pytorch/pytorch/commit/073cee531c9e615a3d0be65ebbb2fe1983f72b20).

With the stack described above I observe failures in one of the benchmark and in number of tests - see errors below. I don't have CUDA, execution goes to CPU with these examples.

```
$ python3 bench/torch_kernels/test_weight_int4pack_mm.py --device cpu
Traceback (most recent call last):
  File "/home/dvrogozh/git/huggingface/optimum-quanto/bench/torch_kernels/test_weight_int4pack_mm.py", line 119, in <module>
    main()
  File "/home/dvrogozh/git/huggingface/optimum-quanto/bench/torch_kernels/test_weight_int4pack_mm.py", line 93, in main
    B_packed = torch._convert_weight_to_int4pack(B_int32, innerKTiles=2)
RuntimeError: _convert_weight_to_int4pack_cpu : expect weight to be kByte.

$ python3 -m pytest --pspec --capture=no test/
FAILED test/tensor/ops/test_linear_dispatch.py::::test_linear_bf16_int4[cpu-bias-256-256-1] - RuntimeError: _convert_weight_to_int4pack_cpu : expect weight to be kByte.
FAILED test/tensor/ops/test_linear_dispatch.py::::test_linear_bf16_int4[cpu-bias-256-256-10] - RuntimeError: _convert_weight_to_int4pack_cpu : expect weight to be kByte.
FAILED test/tensor/ops/test_linear_dispatch.py::::test_linear_bf16_int4[cpu-no-bias-256-256-1] - RuntimeError: _convert_weight_to_int4pack_cpu : expect weight to be kByte.
FAILED test/tensor/ops/test_linear_dispatch.py::::test_linear_bf16_int4[cpu-no-bias-256-256-10] - RuntimeError: _convert_weight_to_int4pack_cpu : expect weight to be kByte.
FAILED test/tensor/optimizers/test_hqq_optimizer.py::::test_hqq_optimizer[cpu-128-first-axis-qint4-bf16-input_shape0] - RuntimeError: _convert_weight_to_int4pack_cpu : expect weight to be kByte.
<...>
```

In a benchmark issue is coming from here:
https://github.com/huggingface/optimum-quanto/blob/601dc193ce0ed381c479fde54a81ba546bdf64d1/bench/torch_kernels/test_weight_int4pack_mm.py#L93

**Comment by dacorvo** [comment]
It looks like the base quantized weights before packing must have a different dtype in pytorch 2.5 (they are int32 in 2.4). This could be addressed with conditional code eventually.

**Comment by dvrogozh** [comment]
@dacorvo : yes, it occurs that pytorch 2.5 has changed input tensor shape from `[n][k] int32` to `[n][k / 2] uint8` - see links below. I have tried to fix this for quanto in #286. Please, help to review.

See: https://github.com/pytorch/pytorch/pull/129940
See: https://github.com/pytorch/pytorch/commit/6f662e95756333284450ff9c3c6e78c796aa6e77
